sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/implementation/TableUtils.java

```diff
@@ -612,16 +613,68 @@ public static String[] getKeysFromToken(String token) {
         return keys;
     }
 
-    public static <T> T callWithOptionalTimeout(Supplier<T> callable, ExecutorService threadPool, Duration timeout,
-        ClientLogger logger) {
+    public static Context skip409Logging(Context context) {
+        return context.addData("skip409logging", true);
+    }
+
+    public static <T> Response<T> callWithOptionalTimeout(Supplier<Response<T>> callable, ExecutorService threadPool, Duration timeout, ClientLogger logger) {
+        try {
+            return callHandler(callable, threadPool, timeout, logger);
+        } catch (Throwable thrown) {
+            Throwable exception = mapThrowableToTableServiceException(thrown);
+            throw logger.logExceptionAsError((RuntimeException) exception);
+        }
+
+    }
+
+    public static <T> Response<T> callWithOptionalTimeoutAndContext(Supplier<Response<T>> callable, ExecutorService threadPool, Duration timeout, ClientLogger logger, Context context) {
+        try {
+            return callHandler(callable, threadPool, timeout, logger);
+        } catch (Throwable thrown) {
+            Throwable exception = mapThrowableToTableServiceException(thrown);
+            context = setContext(context);
+
+            if (exception instanceof TableServiceException) {
+                if (shouldSkip409Logging((TableServiceException) exception, context)) {
+                    // return empty response
+                    HttpResponse resp = ((TableServiceException) exception).getResponse();
+                    return new SimpleResponse<>(resp.getRequest(), resp.getStatusCode(), resp.getHeaders(), null);
+                }
+            }
+
+            throw logger.logExceptionAsError((RuntimeException) exception);
+        }
+    }
+
+    public static <T> PagedIterable<T> callIterableWithOptionalTimeout(Supplier<PagedIterable<T>> callable, ExecutorService threadPool, Duration timeout, ClientLogger logger) {
+        try {
+            return callHandler(callable, threadPool, timeout, logger);
+        } catch (Throwable thrown) {
+            Throwable exception = mapThrowableToTableServiceException(thrown);
+            throw logger.logExceptionAsError((RuntimeException) exception);
+        }
+    }
+
+    private static <T> T callHandler(Supplier<T> callable, ExecutorService threadPool, Duration timeout, ClientLogger logger) throws Throwable {
```

Comment:
We should never check with `Throwable`, at most `Exception`. We want `Error`s to propagate as those are critical issues during runtime

Reply:
@alzimmermsft I can see your point